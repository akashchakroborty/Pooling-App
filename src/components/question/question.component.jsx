import React from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import {
  QuestionContainer,
  QuestionDetails,
  QuestionFooter
} from "./question.styles";

export const Question = ({ question, published_at, choices, url, history }) => {
  return (
    <QuestionContainer>
      <QuestionDetails>
        <h2>{question}</h2>
        <h3>
          Published at : <span>{new Date(published_at).toDateString()}</span>
        </h3>
        <h3>
          Number of choices : <span>{choices.length}</span>
        </h3>
      </QuestionDetails>
      <QuestionFooter>
        <button
          onClick={() => {
            history.push(`${url}`);
          }}
        >
          Vote Now
        </button>
      </QuestionFooter>
    </QuestionContainer>
  );
};

Question.propTypes = {
  question: PropTypes.string.isRequired,
  published_at: PropTypes.string.isRequired,
  choices: PropTypes.array.isRequired,
  url: PropTypes.string.isRequired,
  history: PropTypes.object.isRequired
};

export default withRouter(Question);
