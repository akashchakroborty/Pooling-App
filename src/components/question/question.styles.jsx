import styled from "styled-components";

export const QuestionContainer = styled.div`
  background: #0d223f;
  border: 1px solid #888;
  border-radius: 4px;
  -webkit-box-shadow: 4px 4px 10px -4px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 4px 4px 10px -4px rgba(0, 0, 0, 0.75);
  box-shadow: 4px 4px 10px -4px rgba(0, 0, 0, 0.75);
`;

export const QuestionDetails = styled.div`
  display: flex;
  flex-direction: column;
  padding: 0 15px;
  border-bottom: 1px solid #c7c7c7;
  width: 100%;
  h3 {
    font-weight: normal;
  }
`;

export const QuestionFooter = styled.div`
  text-align: right;
  padding: 15px;
`;
