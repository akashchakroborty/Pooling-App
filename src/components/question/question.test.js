import React from "react";
import { shallow } from "enzyme";
import { Question } from "./question.component";

describe("Question component", () => {
  let mockHistory;
  let wrapper;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      question: "",
      published_at: "",
      choices: [],
      url: "",
      history: mockHistory
    };

    wrapper = shallow(<Question {...mockProps} />);
  });

  it("should render Question component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push when button is clicked", () => {
    wrapper.find("button").simulate("click");
    expect(mockHistory.push).toHaveBeenCalled();
  });
});
