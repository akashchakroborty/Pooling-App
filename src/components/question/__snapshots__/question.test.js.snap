// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component should render Question component 1`] = `ShallowWrapper {}`;
